### PR TITLE
chm-416 financial framework 2015 view report page

### PR DESCRIPTION
**Pull request by liyuanmontreal**
### General description
migrate financial framework 2015 view report page;
section 0-3  are same as financial framework 2020, 
so add a seperate component view-financial-report to reuse it in two reports.

### Designs
same as absch project / old chm project

### Testing instructions

image
image
image
image
image
image
image
image

### CODE REVIEW

File: app/views/forms/view/chm/financial-report-2015.vue [hunk 1]
```diff
@@ -0,0 +1,648 @@
...
+                            <td><strong>{{t("average")}}</strong></td>
+                            <td>{{typeAverageAmount(document.domesticExpendituresData.expenditures,'amount')  | "number:0"}}</td>
+                            <td>{{confidenceAverage(document.domesticExpendituresData.expenditures)}}</td>
+                        </tr>
+                    </tbody>
+                </table>
+                
+                <div v-if="hasValue(document.domesticExpendituresData.govCentralDirectlyRelated)  || hasValue(document.domesticExpendituresData.govCentralIndirectlyRelated) ||
+                            hasValue(document.domesticExpendituresData.provincialDirectlyRelated) || hasValue(document.domesticExpendituresData.provincialIndirectlyRelated) ||
+                            hasValue(document.domesticExpendituresData.municipalDirectlyRelated)  || hasValue(document.domesticExpendituresData.municipalIndirectlyRelated)  ||
```
Comment (tahmeedshah) [inline app/views/forms/view/chm/financial-report-2015.vue:60]: To manage complex v-if conditions in Vue 3, it's a good practice to move the condition logic into a computed property.

File: app/views/forms/view/directives/view-financial-report.vue [hunk 2]
```diff
@@ -0,0 +1,518 @@
...
+    const { t } = useI18n({ messages });
+
+    const props = defineProps({
+        document: { type:Object, required: true },
+        locale  : { type:String, required: true },
+    })  
+
+    const options  = {
+        multipliers : 		[{identifier:'units',	      title: {en:'in units'}},   		   {identifier:'thousands', title: {en:'in thousands'}}, 		{identifier:'millions', title: {en:'in millions'}}],
+        methodology : 		[{identifier:'oecd_dac',      title: {en:'OECD DAC Rio markers'}}, {identifier:'other', 	title: {en:'Other'       }}],
```
Comment (tahmeedshah) [inline app/views/forms/view/directives/view-financial-report.vue:342]: will translation work here ?

Comment (liyuanmontreal) [reply app/views/forms/view/directives/view-financial-report.vue:342]: yes ,it works 
image

File: app/views/forms/view/directives/view-financial-report.vue [hunk 3]
```diff
@@ -0,0 +1,510 @@
...
+    const hasMethodologicalInformationData = computed(()=>{
+        return  props.document.internationalResources.progressData && 
+                (props.document.internationalResources.progressData.odaConfidenceLevel || 
+                props.document.internationalResources.progressData.oofConfidenceLevel  || 
+                props.document.internationalResources.progressData.otherConfidenceLevel)
+    });
+
+    const hasFlowData = computed(()=>{    
+        if (props.type=='2015'){
+            return   props.document.internationalResources.baselineData && props.document.internationalResources.baselineData.baselineFlows;
```
Comment (liyuanmontreal) [inline app/views/forms/view/directives/view-financial-report.vue:412]: props.document?.internationalResources?.baselineData?.baselineFlows;

**Review by blaisefonseca — APPROVED** [review]

**Pull request merged**